packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -587,6 +623,8 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
         defaultProps: {
           className: Dropdown.slotClassNames.item,
           active: highlightedIndex === index,
+          selected: !this.props.multiple ? selectedIndex === index : false,
```

Comment:
do we really need this logic? can't we just say `selected: !this.props.multiple && value === item` ? Even the first check is not needed, as when it's multiple it will never be equality between the item and the array.

Reply:
I am using the `changes.selectedItem` which is always the last selected item (if I use the value I will always have to do checks whether it is array or not, and do casting - which in my opinion is more prone for bugs in the future). I can rewrite this to: `selected: !this.props.multiple && selectedIndex === index`, I agree that it is more readable. Are you ok with this?

Reply:
If it works I am ok, I just thought that simply doing `value === item` will be enough. If it's multiple, it's false. If it's single, it will compare objects and if those are equal then we are good.

Why would you need to enforce array checking?

Reply:
Changed